src/lib.rs

```diff
@@ -157,35 +157,35 @@ impl<T: CoordinateType> PiecewiseLinearFunction<T> {
             Err(idx) => {
                 if idx == 0 || idx == self.coordinates.len() {
                     // Outside the function's domain
-                    return None;
+                    return Err(Error::OutsideDomain);
                 } else {
                     idx
                 }
             }
         };
 
         if idx == 0 {
-            Some(Line::new(self.coordinates[idx], self.coordinates[idx + 1]))
+            Ok(Line::new(self.coordinates[idx], self.coordinates[idx + 1]))
         } else {
-            Some(Line::new(self.coordinates[idx - 1], self.coordinates[idx]))
+            Ok(Line::new(self.coordinates[idx - 1], self.coordinates[idx]))
         }
     }
 
     /// Computes the value f(x) for this piecewise linear function.
     ///
     /// Returns `None` if `x` is outside the domain of f.
```

Comment:
Why aren't we also returning `return Err(Error::OutsideDomain);` in this method?